packages/base/src/autotask/index.ts

```diff
@@ -34,18 +41,26 @@ export abstract class BaseAutotaskClient {
 
   // eslint-disable-next-line @typescript-eslint/ban-types
   protected async execute<T>(request: object): Promise<T> {
-    const result = await this.lambda
+    const invocationTimeStamp = getTimestampInSeconds();
+
+    this.invocationRateLimit.checkRateFor(invocationTimeStamp);
+
+    const invocationRequest = this.lambda
```

Comment:
Let's use await here, as before, to keep things clear and readable.
We could increment rate just before invoking lambda, I don't see big issues with that.